File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/ledger/validations/HouseholdIdPresenceValidator.java
package com.decrux.pocketr.api.services.ledger.validations;

import com.decrux.pocketr.api.exceptions.BadRequestException;
import java.util.UUID;
import org.springframework.stereotype.Component;

@Component
public class HouseholdIdPresenceValidator {

    public UUID validate(UUID householdId) {
        if (householdId == null) {
            throw new BadRequestException("householdId is required for household mode");
        }
        return householdId;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/ledger/validations/HouseholdIdPresenceValidator.kt
package com.decrux.pocketr.api.services.ledger.validations

import com.decrux.pocketr.api.exceptions.BadRequestException
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class HouseholdIdPresenceValidator {
    fun validate(householdId: UUID?): UUID =
        householdId ?: throw BadRequestException("householdId is required for household mode")
}
